### PR TITLE
[video_player] Adds option `loop`,  `volume` and `playAfterInitialized` to VideoPlayerController constructor

### PR DESCRIPTION
**Pull request by vanvixi**
## Feature request: 
- https://github.com/flutter/flutter/issues/134894

## Can be used as follows:
```
final VideoPlayerController controller = VideoPlayerController.networkUrl(
  Uri(),
  isLooping: true,
  volume: 0,
  playAfterInitialized: true,
);
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by tarrinneal — CHANGES_REQUESTED** [review]
You've checked all the boxes on the pr checklist, but haven't done all of the items listed in them. Please try to be more thorough when reading through the checklist in the future.

This will need at least some adjustments to existing tests, if not new tests for this change. 

It seems to me that when this was designed the parameters may have been kept sparse for a reason. Perhaps @stuartmorgan has some insight into this.

**Comment by stuartmorgan** [comment]
> [x] I listed at least one issue that this PR fixes in the description above.

What problem is this PR solving? Please file an issue using the "Feature request" template and filling in the requested information, so that it's clear what the goal of this proposal is.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
(Updating review flag to reflect my comment above.)

File: packages/video_player/video_player/lib/video_player.dart [hunk 1]
```diff
@@ -252,15 +252,22 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
...
-      Future<ClosedCaptionFile>? closedCaptionFile,
-      this.videoPlayerOptions})
-      : _closedCaptionFileFuture = closedCaptionFile,
+  VideoPlayerController.asset(
+    this.dataSource, {
+    this.package,
+    Future<ClosedCaptionFile>? closedCaptionFile,
+    this.videoPlayerOptions,
+    bool isLooping = false,
+    double volume = 1.0,
```
[the remark this answers is not in the document]
Comment (vanvixi) [reply packages/video_player/video_player/lib/video_player.dart:261]: @tarrinneal 
Thank you for your reminder.
I fixed it. I renamed `isLooping` -> `loop`.
Also I added the option: `playAfterInitialized`. What do you think about this option?

**Comment by vanvixi** [comment]
> What problem is this PR solving? Please file an issue using the "Feature request" template and filling in the requested information, so that it's clear what the goal of this proposal is..

@stuartmorgan 
Iam currently developing a video playback application similar to TikTok. During the development process, I need to prepare the next video and configure necessary options such as "loop, volume, playAfterInitialized" in advance. The changes in this pull request (PR) make my application faster, and I believe those who are working on a video application like TikTok will also need these options in the constructor.

**Comment by stuartmorgan** [comment]
@vanvixi Again, please [file an issue](https://github.com/flutter/flutter/issues/new?assignees=&labels=&projects=&template=3_feature_request.yml). As [explained in our docs](https://github.com/flutter/flutter/wiki/Tree-hygiene#overview) an issue is the place for discussion of whether a feature should exist, not a PR.

> The changes in this pull request (PR) make my application faster

Specific details on this sound like good information to include in the issue.

**Comment by vanvixi** [comment]
@stuartmorgan 
I have added an issues here: https://github.com/flutter/flutter/issues/134894

**Comment by stuartmorgan** [comment]
Closing, as we never received a response on the issue about the reason for this change.

**Pull request closed without merging**